llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -1230,7 +1236,92 @@ static bool isUnsupportedAMDGPUAddrspace(Value *Addr) {
   return false;
 }
 
+static bool isUnsupportedSPIRAddrspace(Value *Addr, Function *Func) {
+  Type *PtrTy = cast<PointerType>(Addr->getType()->getScalarType());
+  if (PtrTy->getPointerAddressSpace() != 0) {
+    // Skip SPIR-V built-in varibles
+    StringRef Name;
+    if (auto *CE = dyn_cast<ConstantExpr>(Addr)) {
+      const auto OpCode = CE->getOpcode();
+      if (OpCode == Instruction::GetElementPtr ||
+          OpCode == Instruction::AddrSpaceCast) {
+        Name = CE->getOperand(0)->getName();
+      }
+    } else {
+      Name = Addr->getName();
+    }
+    if (Name.starts_with("__spirv_BuiltIn")) {
+      return true;
+    }
+  } else {
+    // FIXME: Check kernel argments instead of value's name
+    auto Name = Addr->getName();
+    if (Name == "_arg_NumWorkItems" || Name == "_arg_KernelFunc") {
+      return true;
+    }
+  }
+  return false;
+}
+
+static GlobalVariable *GetOrCreateGlobalString(Module &M, StringRef Name,
+                                               StringRef String,
+                                               unsigned AddressSpace) {
+  static std::unordered_map<std::string, GlobalVariable *> GlobalStringMap;
+  GlobalVariable *StringGV = nullptr;
+  if (GlobalStringMap.find(String.str()) != GlobalStringMap.end()) {
+    StringGV = GlobalStringMap[String.str()];
+  } else {
+    auto *Ty =
+        ArrayType::get(Type::getInt8Ty(M.getContext()), String.size() + 1);
+    StringGV = new GlobalVariable(
+        M, Ty, true, GlobalValue::InternalLinkage,
+        ConstantDataArray::getString(M.getContext(), String), Name, nullptr,
+        GlobalValue::NotThreadLocal, AddressSpace);
+    GlobalStringMap[String.str()] = StringGV;
+  }
+  return StringGV;
+}
+
+static void AppendDebugInfoToArgs(Instruction *InsertBefore, Value *Addr,
+                                  SmallVector<Value *> &Args) {
+  auto *M = InsertBefore->getModule();
+  auto &C = InsertBefore->getContext();
+  auto &DB = InsertBefore->getDebugLoc();
```

Comment:
The variable name DB is a little confusing here. Maybe change it to DL or Loc?

Reply:
done